genesyscloud/telephony_providers_edges_phone/resource_genesyscloud_telephony_providers_edges_phone_utils.go

```diff
@@ -0,0 +1,310 @@
+		line := platformclientv2.Line{
+			Name:             &lineName,
+			LineBaseSettings: lineBaseSettings,
+		}
+
+		lineId, err := getLineIdByPhoneId(ctx, pp, d.Id())
+		if err != nil {
+			log.Printf("Failed to retrieve ID for phone %s: %v", d.Id(), err)
+		} else {
+			line.Id = &lineId
+		}
+
+		lines = append(lines, line)
+
+		linesPtr = &lines
+		return
+	}
+
+	for i := 0; i < len(lineStringList); i++ {
+		lineName := "line_" + *lineBaseSettings.Id + "_" + strconv.Itoa(i+1)
+		properties := map[string]interface{}{
+			"station_identity_address": &map[string]interface{}{
+				"value": &map[string]interface{}{
+					"instance": (lineStringList)[i],
+				},
+			},
+		}
+		lines = append(lines, platformclientv2.Line{
+			Name:             &lineName,
+			LineBaseSettings: lineBaseSettings,
+			Properties:       &properties,
+		})
+	}
+
+	linesPtr = &lines
+	isStandAlone = true
+
+	return
+}
+
+func getLineIdByPhoneId(ctx context.Context, pp *phoneProxy, phoneId string) (string, error) {
+	phone, _, err := pp.getPhoneById(ctx, phoneId)
+	if err != nil {
+		return "", err
+	}
+	if phone.Lines != nil && len(*phone.Lines) > 0 {
+		return *(*phone.Lines)[0].Id, nil
+	}
+	return "", fmt.Errorf("could not access line ID for phone %s", phoneId)
+}
+
+func buildSdkCapabilities(d *schema.ResourceData) *platformclientv2.Phonecapabilities {
+	if capabilities := d.Get("capabilities").([]interface{}); capabilities != nil {
+		sdkPhoneCapabilities := platformclientv2.Phonecapabilities{}
+		if len(capabilities) > 0 {
+			if _, ok := capabilities[0].(map[string]interface{}); !ok {
+				return nil
+			}
+			capabilitiesMap := capabilities[0].(map[string]interface{})
+
+			// Only set non-empty values.
+			provisions := capabilitiesMap["provisions"].(bool)
+			registers := capabilitiesMap["registers"].(bool)
+			dualRegisters := capabilitiesMap["dual_registers"].(bool)
+			var hardwareIdType string
+			if checkHardwareIdType := capabilitiesMap["hardware_id_type"].(string); len(checkHardwareIdType) > 0 {
+				hardwareIdType = checkHardwareIdType
+			}
+			allowReboot := capabilitiesMap["allow_reboot"].(bool)
+			noRebalance := capabilitiesMap["no_rebalance"].(bool)
+			noCloudProvisioning := capabilitiesMap["no_cloud_provisioning"].(bool)
+			mediaCodecs := make([]string, 0)
+			if checkMediaCodecs := capabilitiesMap["media_codecs"].([]interface{}); len(checkMediaCodecs) > 0 {
+				for _, codec := range checkMediaCodecs {
+					mediaCodecs = append(mediaCodecs, fmt.Sprintf("%v", codec))
+				}
+			}
+			cdm := capabilitiesMap["cdm"].(bool)
+
+			sdkPhoneCapabilities = platformclientv2.Phonecapabilities{
+				Provisions:          &provisions,
+				Registers:           &registers,
+				DualRegisters:       &dualRegisters,
+				HardwareIdType:      &hardwareIdType,
+				AllowReboot:         &allowReboot,
+				NoRebalance:         &noRebalance,
+				NoCloudProvisioning: &noCloudProvisioning,
+				MediaCodecs:         &mediaCodecs,
+				Cdm:                 &cdm,
+			}
```

Comment:
Same as in the update and create functions, you can set some of these properties directly instead of using variables. Try something like:
```
Provisions: platformclientv2.Bool(capabilitiesMap["provisions"].(bool))
```